tests/haskell_cabal_library_sublibrary_name/package1.cabal

```diff
@@ -7,8 +7,7 @@ executable package1_executable
     hs-source-dirs:   app
     default-language: Haskell2010
     build-depends:
-        Cabal
-        , base
+        base
```

Comment:
What's the reason for this change?

Reply:
The `Cabal` library is not actually needed here and I thought that it was not compatible with asterius.
Because the asterius test failed with:  `Cabal is not a toolchain library`.

But now that I looked into it deeper I think it was a mistake.
We try to use Cabal as a toolchain library because it is in [this list](https://github.com/tweag/rules_haskell/blob/e67eff938b5c580663c1459d52abb26e90658524/haskell/cabal.bzl#L985-L1022), but this list seems wrong in the case of asterius. So maybe we could recover the Cabal package via some other way in this case ?

Reply:
Oh, I see. Thanks for looking into this.
I think we can remove that list and instead use the location information provided by stack [here](https://github.com/tweag/rules_haskell/blob/e67eff938b5c580663c1459d52abb26e90658524/haskell/cabal.bzl#L1160-L1161). IIRC if there is no `location` field in the spec then it is a core package.
Though, I think this is orthogonal to this PR and perhaps better addressed in a separate PR.